io.openems.edge.evcs.spelsberg.smart/src/io/openems/edge/evcs/spelsberg/smart/EvcsSpelsbergSmartImpl.java

```diff
@@ -0,0 +1,343 @@
+
+		addStatusCallback();
+		addPhaseDetectionCallback();
+		addPowerConsumptionCallback();
+		return modbusProtocol;
+	}
+
+	@Override
+	public String debugLog() {
+		return "Status: " + getStatus().getName() + " | " + "Charge Power Limit: "
+				+ getSetChargePowerLimit().orElse(null) + " | " + "Actual Charge Power: " + getChargePowerTotal()
+				+ " | " + "Charged Energy: " + getEnergySessionChannel().value();
+	}
+
+	@Override
+	public void logDebug(String message) {
+		if (this.config.debugMode()) {
+			this.logInfo(this.log, message);
+		}
+	}
+
+	@Override
+	public EvcsPower getEvcsPower() {
+		return this.evcsPower;
+	}
+
+	@Override
+	public int getConfiguredMinimumHardwarePower() {
+		return currentToPower(this.config.minHwCurrent());
+	}
+
+	@Override
+	public int getConfiguredMaximumHardwarePower() {
+		return currentToPower(this.config.maxHwCurrent());
+	}
+
+	@Override
+	public boolean getConfiguredDebugMode() {
+		return this.config.debugMode();
+	}
+
+	@Override
+	public boolean applyChargePowerLimit(int power) throws Exception {
+		setApplyChargePowerLimit(power);
+		return true;
+	}
+
+	@Override
+	public boolean pauseChargeProcess() throws Exception {
+		this.applyChargePowerLimit(0);
+		return true;
+	}
+
+	@Override
+	public boolean applyDisplayText(String text) throws OpenemsException {
+		return false;
+	}
+
+	@Override
+	public int getMinimumTimeTillChargingLimitTaken() {
+		return 30;
+	}
+
+	@Override
+	public ChargeStateHandler getChargeStateHandler() {
+		return this.chargeStateHandler;
+	}
+
+	@Override
+	public void handleEvent(Event event) {
+		if (!this.isEnabled()) {
+			return;
+		}
+		switch (event.getTopic()) {
+		case EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE:
+			writeHandler.run();
+			break;
+		}
+	}
+
+	private void applyConfig(Config config) {
+		this.config = config;
+		this._setChargingType(ChargingType.AC);
+		this._setPhases(Phases.THREE_PHASE.getValue());
+		this._setFixedMinimumHardwarePower(currentToPower(config.minHwCurrent()));
+		this._setFixedMaximumHardwarePower(currentToPower(config.maxHwCurrent()));
+		this._setPowerPrecision(DEFAULT_POWER_RECISION); // 1A steps
+	}
+
+	private Integer currentToPower(Integer current) {
+		return Math.round(current / 1000f) * DEFAULT_VOLTAGE * getPhasesAsInt();
+	}
+
+	private void addStatusCallback() {
+		this.channel(EvcsSpelsbergSmart.ChannelId.CHARGE_POINT_STATE).onSetNextValue(s -> {
+			ChargePointState state = s.asEnum();
+
+			/**
+			 * Maps the EVCS state to a {@link Status}.
+			 */
+			switch (state) {
+			case CHARGING:
+				this._setStatus(Status.CHARGING);
+				break;
+			case NO_PERMISSION:
+			case CHARGING_STATION_RESERVED:
+				this._setStatus(Status.CHARGING_REJECTED);
+				break;
+			case ERROR:
+				this._setStatus(Status.ERROR);
+				break;
+			case NO_VEHICLE_ATTACHED:
+				this._setStatus(Status.NOT_READY_FOR_CHARGING);
+				break;
+			case CHARGING_PAUSED:
+				this._setStatus(Status.READY_FOR_CHARGING);
```

Comment:
This could be a problem when the evcs is working in a cluster.
In the cluster, we try to give each evcs an initial charging power.
Normally we assume that the "CHARGING_PAUSED" State is present because the charge process was rejected by the EMS. Currently we would aslo set it to CHARGING_REJECTED
As this is not the first time it is not that self-explanatory, there could be a slight change on the sates also, but we could always have Evcss, that do no have a separate internal state for a "ChargePause"

Reply:
Ok. I set the openEMS status CHARGING_REJECTED in case the charging process is paused.
If i understand correct this will resolve the possible issues in a cluster?